### PR TITLE
[#IR-296] added support to take DoneStatusMapping input while creation of Retrospective

### PR DESCRIPTION
**Pull request by chahat-arora**

### CODE REVIEW

File: src/app/shared/task-provider/task-provider.component.html [hunk 1]
```diff
@@ -39,13 +39,36 @@
...
             </div>
         </div>
+        <mat-toolbar class="small-toolbar">
+            Tracker Ticket Status Mapping
+        </mat-toolbar>
+        <div class="table-box padded-container" [formGroupName]="taskProviderConfigKey">
+            <div class="layout-row">
+                <mat-form-field class="flex">
+                    <input matInput 
+                    placeholder="Done (Comma Separated)"
```
Comment (himanshu-garg-165) [inline src/app/shared/task-provider/task-provider.component.html]: Change `Done` to `Completed`.

File: src/app/shared/task-provider/task-provider.component.html [hunk 2]
```diff
@@ -39,13 +39,36 @@
...
+                <mat-form-field class="flex">
+                    <input matInput 
+                    placeholder="Done (Comma Separated)" 
+                    [pattern]="commaSeparatedRegex" 
+                    [formControlName]="trackerTicketStatusMap.DONE"/>
+                    <mat-error *ngIf="doneStatusControl.errors && doneStatusControl.errors['pattern']">
+                        It should be a comma separated string
+                    </mat-error> 
+                    <mat-hint> 
+                        <mat-icon>info</mat-icon>
```
Comment (himanshu-garg-165) [inline src/app/shared/task-provider/task-provider.component.html]: Add CSS to beautify the info section, similar to the one done for provider custom fields.

File: src/app/shared/task-provider/task-provider.component.html [hunk 3]
```diff
@@ -39,13 +39,36 @@
...
         </div>
+        <mat-toolbar class="small-toolbar">
+            Tracker Ticket Status Mapping
+        </mat-toolbar>
+        <div class="table-box padded-container" [formGroupName]="taskProviderConfigKey">
+            <div class="layout-row">
+                <mat-form-field class="flex">
+                    <input matInput 
+                    placeholder="Done (Comma Separated)" 
+                    [pattern]="commaSeparatedRegex"
```
Comment (himanshu-garg-165) [inline src/app/shared/task-provider/task-provider.component.html]: Change the regex to handle the following special characters `'`, `-`, `_`.  For example, the regex breaks for `Won't Fix` status for JIRA.

File: src/assets/config/config.dev.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (himanshu-garg-165) [inline src/assets/config/config.dev.json]: Do not commit local config files into the code. Use `config.local.json` to override config for the development environment as mentioned in the README.

File: src/constants/app-constants.ts [hunk 5]
```diff
@@ -282,6 +282,10 @@ export const TRACKER_TICKET_TYPE_MAP = {
     BUG: 'BugTypes',
 };
 
+export const TRACKER_TICKET_STATUS_MAP = {
+    DONE: 'DoneStatus',
```
Comment (himanshu-garg-165) [inline src/constants/app-constants.ts]: Shouldn't we use `DoneType` here instead of `DoneStatus`?

File: src/constants/app-constants.ts [hunk 6]
```diff
@@ -282,6 +282,10 @@ export const TRACKER_TICKET_TYPE_MAP = {
     BUG: 'BugTypes',
 };
 
-export const COMMA_SEPARATED_STRING_PATTERN = '([a-zA-Z\\s]+,\\s?)*([(a-zA-Z\\s]+)';
+export const TRACKER_TICKET_STATUS_MAP = {
+    DONE: 'DoneStatus',
+};
+
+export const COMMA_SEPARATED_STRING_PATTERN = '([a-zA-Z-_\'\\s]+,\\s?)*([(a-zA-Z-_\'\\s]+)';
```
Comment (himanshu-garg-165) [inline src/constants/app-constants.ts]: Fix the regex as discussed.

File: src/app/shared/task-provider/task-provider.component.html (hunk 2 above)
Comment (chahat-arora) [reply src/app/shared/task-provider/task-provider.component.html]: Done

**Pull request merged**